src/d3a/d3a_core/sim_results/stats.py

```diff
@@ -111,27 +114,34 @@ def energy_bills(self, area, past_market_types):
         if not area.children:
             return None
         result = self._get_child_data(area)
-        result["market_fee"] = 0.0
         for market in self._get_past_markets_from_area(area, past_market_types):
-            result["market_fee"] += market.market_fee
             for trade in market.trades:
                 buyer = area_name_from_area_or_iaa_name(trade.buyer)
                 seller = area_name_from_area_or_iaa_name(trade.seller)
                 if buyer in result:
                     self._store_bought_trade(result[buyer], trade.offer)
                 if seller in result:
                     self._store_sold_trade(result[seller], trade.offer)
-
         for child in area.children:
             child_result = self.energy_bills(child, past_market_types)
             if child_result is not None:
-                result[child.name]['market_fee'] = child_result.pop("market_fee")
                 result[child.name]['children'] = child_result
+                result[child.name]['market_fee'] = self.market_fees[child.name]
 
         return result
 
+    def _accumulate_market_fees(self, area, past_market_types):
+        if area.name not in self.market_fees:
+            self.market_fees[area.name] = 0.0
+        for market in self._get_past_markets_from_area(area, past_market_types):
```

Comment:
This only works, if ConstSettings.GeneralSettings.KEEP_PAST_MARKETS is True, right?

Reply:
you are right, the self.market_fees dictionary needs to be reset if the KEEP_PAST_MARKETS is true. Fixing that